packages/aws-rfdk/lib/lambdas/nodejs/lib/configure-spot-event-plugin/test/spot-event-plugin-client.test.ts

```diff
@@ -238,4 +256,163 @@ describe('SpotEventPluginClient', () => {
+      data: {
+        Pools: {},
+      },
+    };
+
+    // WHEN
+    // eslint-disable-next-line dot-notation
+    spotEventPluginClient['deadlineClient'].GetRequest = jest.fn().mockResolvedValue(invalidGroupResponse);
+    // eslint-disable-next-line dot-notation
+    const promise = spotEventPluginClient['getCollection'](CollectionType.Group);
+
+    // THEN
+    await expect(promise).rejects.toThrowError(`Failed to get group collection. Reason: Failed to receive a group collection. Invalid response: ${JSON.stringify(invalidGroupResponse.data)}.`);
+  });
+
+  test.each([
+    [CollectionType.Group, groupsColection],
+    [CollectionType.Pool, poolsColection],
+  ])('successful saveCollection for %s', async (type: CollectionType, expectedResult: any) => {
+    // GIVEN
+    // eslint-disable-next-line dot-notation
+    spotEventPluginClient['deadlineClient'].PostRequest = jest.fn().mockResolvedValue({});
+
+    // WHEN
+    // eslint-disable-next-line dot-notation
+    const result = await spotEventPluginClient['saveCollection'](expectedResult, type);
+
+    // THEN
+    expect(result).toBeTruthy();
+    // eslint-disable-next-line dot-notation
+    expect(spotEventPluginClient['deadlineClient'].PostRequest).toBeCalledTimes(1);
+    expect(consoleLogMock.mock.calls.length).toBe(2);
+    expect(consoleLogMock.mock.calls[0][0]).toMatch(`Saving ${type} collection:`);
+    expect(consoleLogMock.mock.calls[1][0]).toBe(expectedResult);
+  });
+
+  test('failed saveCollection', async () => {
+    // GIVEN
+    const statusMessage = 'error message';
+
+    // eslint-disable-next-line dot-notation
+    spotEventPluginClient['deadlineClient'].PostRequest = jest.fn().mockRejectedValue(new Error(statusMessage));
+
+    // WHEN
+    // eslint-disable-next-line dot-notation
+    const result = await spotEventPluginClient['saveCollection'](groupsColection, CollectionType.Group);
+
+    // THEN
+    expect(result).toBeFalsy();
+    expect(consoleErrorMock.mock.calls.length).toBe(1);
+    expect(consoleErrorMock.mock.calls[0][0]).toMatch(`Failed to save group collection. Reason: ${statusMessage}`);
+  });
+
+  test.each([
+    [groupsColection.Pools,1],
+    [[], 0],
+  ])('successful call addGroup with %s', async (groupsCollection: string[], reuquestsCount: number) => {
+    // GIVEN
+    // eslint-disable-next-line dot-notation
+    spotEventPluginClient['deadlineClient'].GetRequest = jest.fn().mockResolvedValue(successfulGroupResponse);
+    // eslint-disable-next-line dot-notation
+    spotEventPluginClient['deadlineClient'].PostRequest = jest.fn().mockReturnValue(true);
+
+    // WHEN
+    await spotEventPluginClient.addGroups(groupsCollection);
+
+    // THEN
+    // eslint-disable-next-line dot-notation
+    expect(spotEventPluginClient['deadlineClient'].GetRequest).toBeCalledTimes(reuquestsCount);
+
+    // eslint-disable-next-line dot-notation
+    expect(spotEventPluginClient['deadlineClient'].PostRequest).toBeCalledTimes(reuquestsCount);
+  });
```

Comment:
The only thing we are asserting in this test is that `deadlineClient.GetRequest` and `deadlineClent.PostRequest` are being called once. We should add more assertions here to confirm that the correct behavior is being applied. For this, we probably want to split this into a few tests:

*  A test to ensure groups are added if they don't already exist
*  A test to ensure existing groups are not duplicated
*  A test to ensure a mix of (non-)existing groups are correctly merged

Same applies for the `.addPools(...)` test below.

Reply:
Great job. Resolved